receiver/splunkhecreceiver/receiver.go

```diff
@@ -231,20 +250,69 @@ func (r *splunkReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+func (r *splunkReceiver) writeSuccessResponseWithAck(ctx context.Context, resp http.ResponseWriter, eventCount int, channelID string) {
+	if r.ackExt == nil {
+		return
+	}
+
+	ackID := r.ackExt.ProcessEvent(channelID)
+	r.ackExt.Ack(channelID, ackID)
+	r.writeSuccessResponse(ctx, resp, eventCount, []byte(fmt.Sprintf(responseOKWithAckID, ackID)))
+}
+
+func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int, bodyContent []byte) {
 	resp.Header().Set(httpContentTypeHeader, httpJSONTypeHeader)
 	resp.WriteHeader(http.StatusOK)
-	if _, err := resp.Write(okRespBody); err != nil {
+	if _, err := resp.Write(bodyContent); err != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, eventCount, err)
 	}
 }
 
+func (r *splunkReceiver) handleAck(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBodyPostOnly, 0, errInvalidMethod)
+		return
+	}
+
+	// shouldn't run into this case since we only enable this handler IF ackExt exists. But we have this check just in case
+	if r.ackExt == nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, 0, errExtensionMissing)
+		return
+	}
+
+	var channelID string
+	var extracted bool
+	if channelID, extracted = r.extractChannelHeader(req); extracted {
+		if channelErr := r.validateChannelHeader(channelID); channelErr != nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, []byte(channelErr.Error()), 0, channelErr)
+			return
+		}
+	} else {
+		r.failRequest(ctx, resp, http.StatusBadRequest, requiredDataChannelHeader, 0, nil)
+		return
+	}
+
+	dec := json.NewDecoder(req.Body)
+	var ackRequest splunk.AckRequest
+
+	err := dec.Decode(&ackRequest)
+	if err != nil || len(ackRequest.Acks) == 0 {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidFormatRespBody, 0, err)
```

Comment:
you might need to provide an err here so that the metrics report reports an error.

Reply:
not sure what you mean, isn't an error provided as the 6th parameter on line 301?

Reply:
Ah, I see what you mean. I seperated the two cases and added an new error if ackRequest.Acks length == 0

Reply:
great!